### PR TITLE
activate_io fixed to work with post fixed provider_actuators

### PR DESCRIPTION
**Pull request by ddvdd2005**
## Description
Give a brief description of the modifications

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py [hunk 1]
```diff
@@ -1,59 +1,69 @@
...
         State to active the IOs of the submarine (arm not supported yet)
 
-        -- element              uint8       1 : torpedos
-                                            2 : droppers
-                                            3 : arm
-        -- side                 uint8       1 : port
-                                            2 : starboard
+        -- element              uint8       see ActuatorDoAction message
+        -- side                 uint8       see ActuatorDoAction message
+        -- action               uint8       see ActuatorDoAction message
```
Comment (Kam0ss) [inline sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: Leave the actual values needed

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py [hunk 2]
```diff
@@ -1,59 +1,69 @@
...
-        except rospy.ServiceException as exc:
-            rospy.loginfo('Service did not process request : ' + str(exc))
-            return 'failed'
+        self.action_sub = rospy.Subscriber('/provider_actuators/do_action_from_actuators', ActuatorSendReply, self.reply)
+        self.action_pub.publish(ActuatorDoAction(self.element, self.side, self.action))
+        self.time_start = time()
+        self.timeout_pub.publish(missionTimerFunc(f"activate_io (element{self.element}, side{self.side})", self.timeout, str(self.time_start), 1))
+        rospy.loginfo(f'Action : {int(self.element)} is launch')
+
+    def reply(self, data):
```
Comment (Kam0ss) [inline sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: Should be reply_cb just for clarification that it's a callback

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py [hunk 3]
```diff
@@ -1,59 +1,69 @@
...
+                self.timeout_pub.publish(missionTimerFunc("activate_io (element{self.element}, side{self.side})", self.timeout, str(self.time_start), 4))
+                self.response = "failed"
+            elif data.response == ActuatorSendReply.RESPONSE_SUCCESS:
+                self.timeout_pub.publish(missionTimerFunc("activate_io (element{self.element}, side{self.side})", self.timeout, str(self.time_start), 2))
+                self.response = "continue"
+            elif data.response == ActuatorSendReply.RESPONSE_TIMED_OUT:
+                self.timeout_pub.publish(missionTimerFunc("activate_io (element{self.element}, side{self.side})", self.timeout, str(self.time_start), 3))
+                self.response = "timed out"
+            else:
+                rospy.logerr(f"/provider_actuators/do_action_from_actuators sent a message with the response code {data.response} which should not exist")
```
Comment (Kam0ss) [inline sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: Use Logger.log

**Review by Kam0ss — CHANGES_REQUESTED** [review]
Adjustements needed

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py (hunk 1 above)
Comment (Kam0ss) [reply sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: done

Comment (ddvdd2005) [reply sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: done

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py (hunk 2 above)
Comment (ddvdd2005) [reply sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: done

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py (hunk 3 above)
Comment (ddvdd2005) [reply sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: done

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py [hunk 4]
```diff
@@ -1,59 +1,69 @@
...
-        try:
-            self.do_action(self.action_element[str(int(self.param_element))], self.action_side[str(int(self.param_side))], self.action.ACTION_DROPPER_LAUNCH)
-        except rospy.ServiceException as exc:
-            rospy.loginfo('Service did not process request : ' + str(exc))
-            return 'failed'
+        self.action_sub = rospy.Subscriber('/provider_actuators/do_action_from_actuators', ActuatorSendReply, self.reply_cb)
+        self.action_pub.publish(ActuatorDoAction(self.element, self.side, self.action))
+        self.time_start = time()
+        self.timeout_pub.publish(missionTimerFunc(f"activate_io (element{self.element}, side{self.side})", self.timeout, str(self.time_start), 1))
+        rospy.loginfo(f'Action : {int(self.element)} is launch')
```
Comment (Kam0ss) [inline sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: Should be Logger.log

**Review by Kam0ss — CHANGES_REQUESTED** [review]
Just one rospy.loginfo to change to Logger.log

File: sonia_hardware_states/src/sonia_hardware_states/activate_io.py (hunk 4 above)
Comment (ddvdd2005) [reply sonia_hardware_states/src/sonia_hardware_states/activate_io.py]: done

File: sonia_flexbe_behaviors/manifest/test_io.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+	<description>
+		test the io by droping droppers after pulling kill-mission
+	</description>
+
+
+	<!-- Contained Behaviors -->
+
+	<!-- Available Parameters -->
+
+</behavior>
```
Comment (Kam0ss) [inline sonia_flexbe_behaviors/manifest/test_io.xml]: Remove this file

File: sonia_flexbe_behaviors/src/sonia_flexbe_behaviors/test_io_sm.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+										autonomy={'continue': Autonomy.Off, 'failed': Autonomy.Off, 'timeout': Autonomy.Off})
+
+
+		return _state_machine
+
+
+	# Private functions can be added inside the following tags
+	# [MANUAL_FUNC]
+	
+	# [/MANUAL_FUNC]
```
Comment (Kam0ss) [inline sonia_flexbe_behaviors/src/sonia_flexbe_behaviors/test_io_sm.py]: Remove this file

**Review by Kam0ss — CHANGES_REQUESTED** [review]
Remove tests files

**Review by Kam0ss — APPROVED** [review]

**Pull request merged**